src/components/Missions.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import * as api from '../api/api';
+import addMission from '../redux/actions/missions';
+import MissionCard from './MissionCard';
+import './Missions.css';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const missionList = useSelector((state) => state.missions);
+
+  const getMissions = () => (dispatch) => {
+    api.fetchMissions()
+      .then((data) => data.forEach((mission) => {
+        const allowed = ['mission_id', 'mission_name', 'description'];
+        const newMission = Object.keys(mission)
+          .filter((key) => allowed.includes(key))
+          .reduce((obj, key) => {
+            const temp = obj;
+            temp[key] = mission[key];
+            return temp;
+          }, {});
+        dispatch(addMission(newMission));
+      }));
+  };
+
+  useEffect(() => {
+    if (!missionList.length) {
+      dispatch(getMissions());
+    }
+  }, []);
+
+  const myMissionsArr = missionList.map((missions) => (
+    <MissionCard
+      key={missions.mission_id}
+      missionid={missions.mission_id}
+      name={missions.mission_name}
+      description={missions.description}
+      reserved={missions.reserved}
+    />
+  ));
```

Comment:
[OPTIONAL] We could make this block DRYer by destructuring `missions` when we map. Then we don't have to repeat `missions` each time